File: src/main/java/me/chriss99/glabstractions/IntGLBuffer.java
package me.chriss99.glabstractions;

import static org.lwjgl.opengl.GL11.GL_INT;
import static org.lwjgl.opengl.GL15.glBufferData;

public class IntGLBuffer extends GLBuffer {
    public IntGLBuffer(int target) {
        super(target, GL_INT);
    }

    public IntGLBuffer updateData(int[] data, int usage) {
        bind();
        glBufferData(target, data, usage);
        return this;
    }
}

File: src/main/kotlin/glabstractions/IntGLBuffer.kt
package me.chriss99.glabstractions

import org.lwjgl.opengl.GL11.GL_INT
import org.lwjgl.opengl.GL15.glBufferData

class IntGLBuffer(target: Int) : GLBuffer(target, GL_INT) {
    fun updateData(data: IntArray, usage: Int): IntGLBuffer {
        bind()
        glBufferData(target, data, usage)
        return this
    }
}